web-services-app/src/main/java/edu/unc/lib/boxc/web/services/rest/modify/ThumbnailController.java

```diff
@@ -0,0 +1,114 @@
+ * assigning a child object to use as a thumbnail
+ *
+ * @author lfarrell
+ *
+ */
+@Controller
+public class ThumbnailController {
+    private static final Logger log = LoggerFactory.getLogger(ThumbnailController.class);
+
+    @Autowired
+    private ImportThumbnailService service;
+    @Autowired
+    private ThumbnailRequestSender thumbnailRequestSender;
+    @Autowired
+    private AccessControlService aclService;
+    @Autowired
+    private RepositoryObjectLoader repositoryObjectLoader;
+
+    @PostMapping(value = "edit/displayThumbnail/{pid}")
+    public @ResponseBody
+    ResponseEntity<Object> ImportThumbnail(@PathVariable("pid") String pid,
+                                                     @RequestParam("file") MultipartFile thumbnailFile)
+            throws Exception {
+
+        AgentPrincipals agent = AgentPrincipalsImpl.createFromThread();
+        String mimeType = thumbnailFile.getContentType();
+
+        Map<String, Object> result = new HashMap<>();
+        result.put("action", "editThumbnail");
+        result.put("username", agent.getUsername());
+
+        try (InputStream importStream = thumbnailFile.getInputStream()) {
+            service.run(importStream, agent, pid, mimeType);
+        } catch (IOException e) {
+            log.error("Failed to get submitted file", e);
+            result.put("error", e.getMessage());
+            return new ResponseEntity<>(result, HttpStatus.INTERNAL_SERVER_ERROR);
+        } catch (IllegalArgumentException e) {
+            log.error("Error queueing the job", e);
+            result.put("error", e.getMessage());
+            return new ResponseEntity<>(result, HttpStatus.UNSUPPORTED_MEDIA_TYPE);
+        }
+
+        result.put("destination", pid);
+
+        return new ResponseEntity<>(result, HttpStatus.OK);
+    }
+
+    @PutMapping(value = "/assignThumbnail/{pidString}")
+    @ResponseBody
+    public ResponseEntity<Object> AssignThumbnail(@PathVariable("pidString") String pidString) {
+        PID pid = PIDs.get(pidString);
+
+        AccessGroupSet principals = getAgentPrincipals().getPrincipals();
+        aclService.assertHasAccess("Insufficient permissions to download access copy for " + pidString,
+                pid, principals, Permission.editDescription);
+
+        var object = repositoryObjectLoader.getRepositoryObject(pid);
+        if (!(object instanceof FileObject)) {
+            log.error("Error object is not a file: {}", pidString);
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+
+        var agent = AgentPrincipalsImpl.createFromThread();
+        var request = new ThumbnailRequest();
+        request.setAgent(agent);
+        request.setFilePidString(pidString);
+        try {
+            thumbnailRequestSender.sendToQueue(request);
+        } catch (IOException e) {
+            log.error("Error assigning file {} as thumbnail: {}", request.getFilePidString(), e);
```

Comment:
```suggestion
            log.error("Error assigning file {} as thumbnail", request.getFilePidString(), e);
```
when an exception is provided as the last parameter, the logger automatically formats it into message as the message + stacktrace.